README.md
Qasim (القاسم)
==============

Qasim is a tool born to make your remote shares easily available from the system
tray, on your desktop !

It uses FUSE filesystems, thus everything remains accessible even under
command-line shell, under your favorite operating system (Linux, Windows,
MacOsX).


Requirements
------------

Qasim require the ``qtbindings`` gems. It installs automatically along qasim's
installation, but is quite long to build a native gem package.

**N.B :** If you experience trouble with Qasim dues to Qt bindings, make sure
your Ruby installation was compiled with the ``--enable-shared`` option. 

If it was not, then reinstall it (let's say version 1.9.3-p392) with with :

    $ CONFIGURE_OPTS="--enable-shared" rbenv install 1.9.3-p392


Installation
------------

Simply run :

    $ gem install qasim


Usage
-----

Qasim provide two tools the CLI and the GUI.

To run the GUI :

    $ qasim-gui

Then click on the icon red-yellow-blue icon in the system tray.


Contributing
------------

1. Fork it ( http://github.com/glenux/qasim/fork )
2. Create your feature branch (`git checkout -b my-new-feature`)
3. Commit your changes (`git commit -am 'Add some feature'`)
4. Push to the branch (`git push origin my-new-feature`)
5. Create new Pull Request


Copyright & License
-------------------

Copyright (C) 2010-2014 Glenn Y. Rolland

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.


Alternatives
------------

* [Mountoid](http://kde-apps.org/content/show.php/Mountoid?content=115943)



debian/changelog
qasim (0.1.10) unstable; urgency=medium

  * qasim-gui: update project dependencies upon ruby & gems.
  * Added base files for Gem packaging & distribution.
  * Makefile: fix installation rules for icons & ui library.
  * ui/preferences: Changed class name MapDialog -> Preferences.
  * ui: Changed window icons.
  * Added base for preferences dialog.
  * Added about dialog.
  * data: Move all UI text as resource files.

 -- Glenn Y. Rolland <glenux@glenux.net>  Fri, 09 May 2014 07:31:16 +0200

qasim (0.1.9) unstable; urgency=low

  * qasim-gui: Use /proc/ instead of kill to detect other processes. 

 -- Glenn Y. Rolland <glenux@glenux.net>  Tue, 18 Oct 2011 16:45:31 +0200

qasim (0.1.8) unstable; urgency=low

  * qasim-gui: Another fix in locker process liveness testing 

 -- Glenn Y. Rolland <glenux@glenux.net>  Fri, 16 Sep 2011 12:08:58 +0200

qasim (0.1.7) unstable; urgency=low

  * qasim-gui: test locker process liveness

 -- Glenn Y. Rolland <glenux@glenux.net>  Sat, 10 Sep 2011 13:58:21 +0200

qasim (0.1.6) unstable; urgency=low

  * qasim-gui: handle QIcon::fromTheme NoMethodError exception when needed.
  * qasim-gui: added single instance lock.

 -- Glenn Y. Rolland <glenux@glenux.net>  Wed, 07 Sep 2011 01:08:48 +0200

qasim (0.1.5) unstable; urgency=low

  * map: Added missing require for fileutils. 

 -- Glenn ROLLAND <glenux@fr.st>  Tue, 09 Aug 2011 22:46:33 +0200

qasim (0.1.4) unstable; urgency=low

  * Fixed localpath bug.

 -- Glenn ROLLAND <glenux@fr.st>  Tue, 09 Aug 2011 22:35:10 +0200

qasim (0.1.3) unstable; urgency=low

  * Create mountpoint if non-existant. 

 -- Glenn Y. Rolland <glenux@glenux.net>  Sun, 07 Aug 2011 17:43:56 +0200

qasim (0.1.2) unstable; urgency=low

  * Disabled rdebug & added another missing dependency.

 -- Glenn Y. Rolland <glenux@glenux.net>  Sun, 07 Aug 2011 17:06:15 +0200

qasim (0.1.1) unstable; urgency=low

  * Added missing dependencies (libqt4-ruby & ruby)

 -- Glenn Y. Rolland <glenux@glenux.net>  Sun, 07 Aug 2011 16:53:20 +0200

qasim (0.1) unstable; urgency=low

  * Initial version.

 -- Glenn Y. Rolland <glenux@glenux.net>  Fri, 29 Jul 2011 17:01:24 +0200


mapparser.rb
#
# DO NOT MODIFY!!!!
# This file is automatically generated by racc 1.4.5
# from racc grammer file "mapparser.y".
#

require 'racc/parser'



# calc.rb : generated by racc
#
require 'strscan'
#require 'map.rb'


class MapParser < Racc::Parser

module_eval <<'..end mapparser.y modeval..id78efd83f53', 'mapparser.y', 68

	def parse( str )
		puts "parse start..."
		tokens = []

		scanner = StringScanner.new( str )
		puts ( "scanner?" + scanner.string )
		
		until scanner.eos?
			puts "scanning.. at #{scanner.pos}"
			case
				when m = scanner.scan( /\/\/.*$/ )
					# comments
				when m = scanner.scan( /(\s+|\n)/ )
					# whitespace and newlines
				when m = scanner.scan( /\{/ )
					tokens.push [:OBRACE, m]
				when m = scanner.scan( /\}/ )
					tokens.push [:EBRACE, m]
				when m = scanner.scan( /;/ )
					tokens.push [:SEMICOLON, m]
				when m = scanner.scan( /\d+/ )
					tokens.push [:NUMBER, m]
				when m = scanner.scan( /user/i )
					tokens.push [:USER, m]
				when m = scanner.scan( /automount/i )
					tokens.push [:AUTOMOUNT, m]
				when m = scanner.scan( /reconnect/i )
					tokens.push [:RECONNECT, m]
				when m = scanner.scan( /compress/i )
					tokens.push [:COMPRESS, m]
				when m = scanner.scan( /port/i )
					tokens.push [:PORT, m]
				when m = scanner.scan( /(true|1|yes)/i )
					tokens.push [:BOOL_TRUE, m]
				when m = scanner.scan( /(false|0|no)/i )
					tokens.push [:BOOL_FALSE, m]
				when m = scanner.scan( /".+?"/i )
					tokens.push [:STRING, m]
				when m = scanner.scan( /map/i )
					tokens.push [:MAP, m]
				when m = scanner.scan( /server/i )
					tokens.push [:SERVER, m]
				when m = scanner.scan( /./ )
					tokens.push [:UNMATCHED, m]
					puts "UNMATCHED #{m} after #{scanner.pre_match}"
			end
		end
		puts "tokenization ok"
		tokens.push [false, false]
		@last_value = yyparse( tokens, :each )
		puts "parse end... "
	end

	def on_error(error_token_id, error_value, value_stack)
			msg = "parse error "
			msg << "after #{value_stack.last} " if value_stack.length > 1
			#msg << "after #{value_stack.last} " unless value_stack.empty?
			msg << "on #{token_to_str(error_token_id)} #{error_value}"
			raise ParseError, msg
	end


..end mapparser.y modeval..id78efd83f53

##### racc 1.4.5 generates ###

racc_reduce_table = [
 0, 0, :racc_error,
 3, 20, :_reduce_1,
 0, 20, :_reduce_2,
 1, 21, :_reduce_none,
 1, 21, :_reduce_none,
 5, 23, :_reduce_5,
 4, 22, :_reduce_none,
 3, 24, :_reduce_none,
 0, 24, :_reduce_none,
 1, 25, :_reduce_9,
 1, 25, :_reduce_10,
 1, 25, :_reduce_11,
 1, 25, :_reduce_12,
 1, 25, :_reduce_13,
 1, 25, :_reduce_14,
 5, 31, :_reduce_none,
 3, 32, :_reduce_none,
 0, 32, :_reduce_none,
 2, 26, :_reduce_18,
 2, 27, :_reduce_19,
 2, 28, :_reduce_20,
 2, 29, :_reduce_21,
 2, 30, :_reduce_22,
 1, 33, :_reduce_23,
 1, 33, :_reduce_24 ]

racc_reduce_n = 25

racc_shift_n = 44

racc_action_table = [
    25,    33,    34,    28,     3,    16,    17,     5,    18,    20,
    22,     6,    41,    33,    34,    30,    42,    33,    34,    15,
    12,    13,    31,    11,    35,    10,     9,    38,    39,     8,
    43 ]

racc_action_check = [
    14,    18,    18,    14,     1,    14,    14,     1,    14,    14,
    14,     1,    40,    22,    22,    16,    40,    20,    20,    13,
     9,    10,    17,     7,    19,     6,     5,    28,    38,     3,
    42 ]

racc_action_pointer = [
   nil,     4,   nil,    29,   nil,    22,    20,    21,   nil,    15,
    13,   nil,   nil,    13,    -6,   nil,    11,     9,   -16,    22,
     0,   nil,    -4,   nil,   nil,   nil,   nil,   nil,    23,   nil,
   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,    23,   nil,
     6,   nil,    28,   nil ]

racc_action_default = [
    -2,   -25,    -3,   -25,    -4,   -25,   -25,   -25,    44,   -25,
   -25,    -1,    -8,   -25,   -25,    -6,   -25,   -25,   -25,   -25,
   -25,    -9,   -25,   -10,   -11,    -5,   -12,   -13,   -25,   -14,
   -18,   -19,   -20,   -23,   -24,    -7,   -21,   -22,   -25,   -17,
   -25,   -15,   -25,   -16 ]

racc_goto_table = [
    32,    23,    36,     2,    37,     4,    14,    19,    21,     7,
    24,    26,    27,    29,    40,     1 ]

racc_goto_check = [
    14,     8,    14,     3,    14,     4,     5,     6,     7,     2,
     9,    10,    11,    12,    13,     1 ]

racc_goto_pointer = [
   nil,    15,     8,     2,     4,    -6,    -7,    -6,   -13,    -4,
    -3,    -2,    -1,   -25,   -18 ]

racc_goto_default = [
   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,   nil,
   nil,   nil,   nil,   nil,   nil ]

racc_token_table = {
 false => 0,
 Object.new => 1,
 :SEMICOLON => 2,
 :SERVER => 3,
 :STRING => 4,
 :OBRACE => 5,
 :EBRACE => 6,
 :GLOBAL => 7,
 :global_opts => 8,
 :MAP => 9,
 :server_map_opt => 10,
 :USER => 11,
 :PORT => 12,
 :NUMBER => 13,
 :AUTOMOUNT => 14,
 :COMPRESS => 15,
 :RECONNECT => 16,
 :BOOL_TRUE => 17,
 :BOOL_FALSE => 18 }

racc_use_result_var = true

racc_nt_base = 19

Racc_arg = [
 racc_action_table,
 racc_action_check,
 racc_action_default,
 racc_action_pointer,
 racc_goto_table,
 racc_goto_check,
 racc_goto_default,
 racc_goto_pointer,
 racc_nt_base,
 racc_reduce_table,
 racc_token_table,
 racc_shift_n,
 racc_reduce_n,
 racc_use_result_var ]

Racc_token_to_s_table = [
'$end',
'error',
'SEMICOLON',
'SERVER',
'STRING',
'OBRACE',
'EBRACE',
'GLOBAL',
'global_opts',
'MAP',
'server_map_opt',
'USER',
'PORT',
'NUMBER',
'AUTOMOUNT',
'COMPRESS',
'RECONNECT',
'BOOL_TRUE',
'BOOL_FALSE',
'$start',
'commands',
'command_blk',
'global_blk',
'server_blk',
'server_opts',
'server_opt',
'user_opt',
'port_opt',
'automount_opt',
'compress_opt',
'reconnect_opt',
'server_map_blk',
'server_map_opts',
'boolean']

Racc_debug_parser = false

##### racc system variables end #####

 # reduce 0 omitted

module_eval <<'.,.,', 'mapparser.y', 5
  def _reduce_1( val, _values, result )
 puts "GOT A COMMAND"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 6
  def _reduce_2( val, _values, result )
 puts "no command found"
   result
  end
.,.,

 # reduce 3 omitted

 # reduce 4 omitted

module_eval <<'.,.,', 'mapparser.y', 13
  def _reduce_5( val, _values, result )
 puts "[server #{val[1]}]"
   result
  end
.,.,

 # reduce 6 omitted

 # reduce 7 omitted

 # reduce 8 omitted

module_eval <<'.,.,', 'mapparser.y', 23
  def _reduce_9( val, _values, result )
 puts "=> user opt"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 24
  def _reduce_10( val, _values, result )
 puts "=> port opt"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 25
  def _reduce_11( val, _values, result )
 puts "=> automount opt"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 26
  def _reduce_12( val, _values, result )
 puts "=> compress opt"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 27
  def _reduce_13( val, _values, result )
 puts "=> reconnect opt"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 28
  def _reduce_14( val, _values, result )
 puts "=> server_map blk"
   result
  end
.,.,

 # reduce 15 omitted

 # reduce 16 omitted

 # reduce 17 omitted

module_eval <<'.,.,', 'mapparser.y', 39
  def _reduce_18( val, _values, result )
 puts "* user = #{val[1]}"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 42
  def _reduce_19( val, _values, result )
 puts "* port = #{val[1]}"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 45
  def _reduce_20( val, _values, result )
 puts "* automount = #{@bool_result}"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 48
  def _reduce_21( val, _values, result )
 puts "* compress = #{@bool_result}"
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 51
  def _reduce_22( val, _values, result )
 puts ( "* reconnect = #{@bool_result}" )
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 54
  def _reduce_23( val, _values, result )
 @bool_result = true
   result
  end
.,.,

module_eval <<'.,.,', 'mapparser.y', 55
  def _reduce_24( val, _values, result )
 @bool_result = false
   result
  end
.,.,

 def _reduce_none( val, _values, result )
  result
 end

end   # class MapParser


puts "GOO"

bigstr = ""
STDIN.read.split(/\n/).each do |a|
   puts "READING #{a}"
   bigstr = bigstr + a
end

puts "PARSING NOW..."

parser = MapParser.new
begin
	val = parser.parse( bigstr )
	print '= ', val, "\n"
	rescue ParseError
	puts $!
end
